File: DAY100~199/DAY117-programmers-네트워크/younghoon.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean networks[] = new boolean[n];
        for(int i = 0; i < n; i++){
            networks[i] = false;
        }
        for (int i = 0; i < n; i++) {
            if (!networks[i]) {
                dfs(computers, i, networks);
                answer++;
            }
        }
        return answer;
    }
    
    public void dfs(int[][] computers, int i, boolean[] networks) {
        networks[i] = true;
        for (int j = 0; j < computers.length; j++) {
           if (i != j && computers[i][j] == 1 && networks[j] == false) {
                dfs(computers, j, networks);
           }
        }
     }
}

File: DAY001~099/DAY72-programmers-순위/younghoon.py
def solution(n, results):
    win = {x:set() for x in range(1, n+1)}
    lose = {x:set() for x in range(1, n+1)}
    for winner, loser in results:
        win[winner].add(loser)
        lose[loser].add(winner)
    for i in range(1, n+1):
        for winner in lose[i]:
            win[winner].update(win[i])
        for loser in win[i]:
            lose[loser].update(lose[i])
    answer = 0
    for i in range(1, n+1):
        if len(win[i]) + len(lose[i]) == n-1:
            answer += 1
    return answer
